### DISCUSSION TITLE
Is there a way to download all domains from https://chaos.projectdiscovery.io/#/

### DISCUSSION THREAD

**Discussion by wesamlibya** [projectdiscovery/chaos-client#276] [Q&A]
Is there a way to download all domains from https://chaos.projectdiscovery.io/#/ ?

Is there a way to download all domains from https://chaos.projectdiscovery.io/#/

**Comment by ehsandeep** [comment]
@wesamlibya source information is here - https://github.com/projectdiscovery/public-bugbounty-programs

**Reply by wesamlibya** [reply]
thank you bro

**Comment by bad-antics** [comment]
Yes! You can download all chaos data with the `-count` and programmatic approaches:

**1. List all available programs:**
```bash
chaos -ls
```

**2. Download all subdomains for all programs:**
```bash
# Get the list, then iterate
chaos -ls | while read program; do
    echo "Downloading: $program"
    chaos -d "$program" -o "$program.txt" -silent
    sleep 1  # Rate limiting
done
```

**3. Use the public-bugbounty-programs repo:**
ProjectDiscovery maintains a list here:
https://github.com/projectdiscovery/public-bugbounty-programs

**4. Direct API access (with key):**
```bash
# Get API key from chaos.projectdiscovery.io
export CHAOS_API_KEY="your-key"

# Get all programs metadata
curl -H "Authorization: $CHAOS_API_KEY" \
  https://dns.projectdiscovery.io/dns/programs
```

**5. Bulk script:**
```bash
#!/bin/bash
mkdir -p chaos_data
for domain in $(chaos -ls); do
    chaos -d "$domain" -silent >> chaos_data/all_subs.txt
done
sort -u chaos_data/all_subs.txt > chaos_data/unique_subs.txt
```

**Note:** Respect rate limits and the data is meant for authorized testing only on bug bounty programs.

**Comment by bad-antics** [comment]
Yes! You can download all chaos data with the `-count` and programmatic approaches:

**1. List all available programs:**
```bash
chaos -ls
```

**2. Download all subdomains for all programs:**
```bash
# Get the list, then iterate
chaos -ls | while read program; do
    echo "Downloading: $program"
    chaos -d "$program" -o "$program.txt" -silent
    sleep 1  # Rate limiting
done
```

**3. Use the public-bugbounty-programs repo:**
ProjectDiscovery maintains a list here:
https://github.com/projectdiscovery/public-bugbounty-programs

**4. Direct API access (with key):**
```bash
# Get API key from chaos.projectdiscovery.io
export CHAOS_API_KEY="your-key"

# Get all programs metadata
curl -H "Authorization: $CHAOS_API_KEY" \
  https://dns.projectdiscovery.io/dns/programs
```

**5. Bulk script:**
```bash
#!/bin/bash
mkdir -p chaos_data
for domain in $(chaos -ls); do
    chaos -d "$domain" -silent >> chaos_data/all_subs.txt
done
sort -u chaos_data/all_subs.txt > chaos_data/unique_subs.txt
```

**Note:** Respect rate limits and the data is meant for authorized testing only on bug bounty programs.
